src/auth/models.hpp

```diff
@@ -119,18 +120,43 @@ bool operator==(const FineGrainedAccessPermissions &first, const FineGrainedAcce
 
 bool operator!=(const FineGrainedAccessPermissions &first, const FineGrainedAccessPermissions &second);
 
+class FineGrainedAccessHandler final {
+ public:
+  FineGrainedAccessHandler(const FineGrainedAccessPermissions &labelPermissions = {},
+                           const FineGrainedAccessPermissions &edgeTypePermissions = {});
+
+  const FineGrainedAccessPermissions &label_permissions() const;
+  FineGrainedAccessPermissions &label_permissions();
+
+  const FineGrainedAccessPermissions &edge_type_permissions() const;
+  FineGrainedAccessPermissions &edge_type_permissions();
```

Comment:
Is it okay to leave it as it is for now and fix this and Permission class in another task because there are some functions that are used by Permission class and my classes so if I change my class I will need to change everything and I believe that is not in scope of this task?